### PR TITLE
Update ansible and Dockerfile for tbb

### LINKED ISSUE #3485: [functional testing] Install latest Tor Browser, geckodriver in dev + test
We should install latest TB and geckodriver in the dev and test environments. 

Epic: #1569

### PR DESCRIPTION
**Pull request by kushaldas**
## Status

Ready for review.

## Description of Changes

Fixes #3485

Changes proposed in this pull request:

Tor Browser 7.5.6 and geckodriver v0.17.0

## Testing

Tests will fail for now in the whole tbb side.

### CODE REVIEW

File: install_files/ansible-base/roles/app-test/tasks/install_tbb.yml [hunk 1]
```diff
@@ -56,3 +56,16 @@
...
+- name: Download geckodriver for compatibility with Tor
+  get_url:
+    dest: "/opt/geckodriver-v0.17.0-linux64.tar.gz"
+    url: https://github.com/mozilla/geckodriver/releases/download/v0.17.0/geckodriver-v0.17.0-linux64.tar.gz
+    sha256sum: 3154274c050d724eb2f4e8986a58ed37c0138b48304692bf7eeed827a5e82319
+  tags:
+    - apt
+
+- name: install geckodriver
+  shell: |
```
Comment (heartsucker) [inline install_files/ansible-base/roles/app-test/tasks/install_tbb.yml]: You should use the `unarchive` module to un-`tar` the download.

File: securedrop/Dockerfile [hunk 2]
```diff
@@ -20,6 +20,26 @@ RUN curl -LO https://launchpad.net/~ubuntu-mozilla-security/+archive/ubuntu/ppa/
     dpkg -i firefox*deb && apt-get install -f && \
     paxctl -cm /usr/lib/firefox/firefox
 
+RUN apt-get install -y wget libasound2 libdbus-glib-1-2 libgtk2.0-0 libfontconfig1 libxrender1
```
Comment (msheiny) [inline securedrop/Dockerfile]: Can you add these components to the other layer? (line 9). It's best practice to keep these things bundled. You could separate by a comment as a visual separator.

File: securedrop/Dockerfile [hunk 3]
```diff
@@ -20,6 +20,26 @@ RUN curl -LO https://launchpad.net/~ubuntu-mozilla-security/+archive/ubuntu/ppa/
...
+
+RUN gpg --keyserver=hkp://ipv4.pool.sks-keyservers.net --recv-keys EF6E286DDA85EA2A4BA7DE684E2C6E8793298290 && \
```
Comment (msheiny) [inline securedrop/Dockerfile]: I would honestly prefer to drop this network outreach and add the key to version control and then docker `COPY` it into the container. We constantly battle with flakiness of keyservers in various projects

File: securedrop/Dockerfile [hunk 4]
```diff
@@ -20,6 +20,26 @@ RUN curl -LO https://launchpad.net/~ubuntu-mozilla-security/+archive/ubuntu/ppa/
...
+    wget  https://www.torproject.org/dist/torbrowser/7.5.6/tor-browser-linux64-7.5.6_en-US.tar.xz && \
+    wget https://www.torproject.org/dist/torbrowser/7.5.6/tor-browser-linux64-7.5.6_en-US.tar.xz.asc && \
+    gpg --verify tor-browser-linux64-7.5.6_en-US.tar.xz.asc tor-browser-linux64-7.5.6_en-US.tar.xz
+
+RUN tar -xvJf tor-browser-linux64-7.5.6_en-US.tar.xz && \
+    mkdir -p /root/.local/tbb && mv tor-browser_en-US /root/.local/tbb &&\
+    paxctl -cm /root/.local/tbb/tor-browser_en-US/Browser/firefox && \
+    paxctl -cm /root/.local/tbb/tor-browser_en-US/Browser/plugin-container
+
+ENV GECKODRIVER_CHECKSUM=3154274c050d724eb2f4e8986a58ed37c0138b48304692bf7eeed827a5e82319
```
Comment (msheiny) [inline securedrop/Dockerfile]: Can you make this a build `ARG` instead? Allow over-riding outside of this file

Comment (kushaldas) [reply securedrop/Dockerfile]: @msheiny Why do we need to override the checksum?

Comment (msheiny) [reply securedrop/Dockerfile]: oooooooh okay thats not gonna change? i dunno... i just always try to move any unique strings like this outside of the dockerfile.. but i see your point

File: securedrop/Dockerfile [hunk 5]
```diff
@@ -20,6 +21,27 @@ RUN curl -LO https://launchpad.net/~ubuntu-mozilla-security/+archive/ubuntu/ppa/
     dpkg -i firefox*deb && apt-get install -f && \
     paxctl -cm /usr/lib/firefox/firefox
 
+
+COPY ./tor_project_public.pub /opt/
```
Comment (conorsch) [inline securedrop/Dockerfile]: Where is this file? Pulling down the branch locally, I can't find it anywhere. Was it omitted from an add? Recall that we already have the tor signing key at `install_files/ansible-base/roles/tor-hidden-services/files/tor-signing-key.pub`, although we'll have to consider the build context for the image and whether that's accessible here.

**Comment by conorsch** [comment]
> kushal mentioned waiting for upstream changes for CI.

@msheiny Can you elaborate a bit on this? Unclear to me why we'd wait to merge, particularly since this is landing in a feature branch, not `develop`.

> Im happy with his changes so far to this PR

I concur! Modulo a single clarification on the tor apt pubkey (see requested changes above).

File: securedrop/Dockerfile (hunk 5 above)
Comment (kushaldas) [reply securedrop/Dockerfile]: Uh, yes, I added that in the `tbb_in_dev` branch (after I figured that I am missing that) :(

**Comment by redshiftzero** [comment]
In sprint planning today, there was a lot of confusion about what the base branch is here, which branch is the right/current one, and what the workflow here is, so I'm going to write out what I think the workflow should be (following the approach we took for the alembic and journalist GUI feature branch work).

1. `tbb-0.9.0` is the base feature branch for the functional testing work. It is not merged into `develop` until all tickets related to the feature are completed. 
2. PRs implementing tickets for the feature should be proposed here for merge into `tbb-0.9.0` as they are completed. There should be one PR per ticket. If the PR is from a fork, the (checked by default) checkbox indicating that maintainers should be allowed to push commits is enabled, this way others can help the PR along if needed. 
3. If two tickets for a given feature are dependent on one another, and the (first) dependent ticket is not yet merged, work should proceed regardless, by pulling the dependent commits into the branch where work is being done for the second ticket. PRs should be done as the work for a ticket is ready for review. The PR author should indicate in the PR description that the PR should not be merged until the PR for the dependent ticket(s) is merged. 
4. Once all tickets for the feature are implemented, a PR is then proposed for final merge of `tbb-0.9.0` into `develop`. 

Does that seem reasonable @kushaldas @msheiny @conorsch? Please let me know if there is a particular issue with this workflow and we can revise it.

**Comment by kushaldas** [comment]
@conorsch I cherry-picked the commit which adds the missing tor public key. Rest of the branches I am working on are dependent on this branch. This is ready to be merged in the `tbb-0.9.0` branch.

**Comment by msheiny** [comment]
@conorsch can you take another pass here? I dismissed your review since @kushaldas addressed your issue but i wanted to wait to see if you had additional concerns.

**Pull request merged**